File: iosApp/iosApp/Presentation/Main/CafeManagement/CafeDashboard/CafeDashboardEvent.swift
//
//  CafeDashboardEvent.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/10.
//

import Foundation

enum CafeDashboardEvent {
    case navigateBack
    case navigateToBanner(cafeId: String)
    case navigateToBannerEdit(cafeId: String)
    case navigateToCafeInfoEdit(cafeId: String)
    case navigateToNoticeEvent(cafeId: String)
    case navigateToMenuGoods(cafeId: String)
    case navigateToCastEdit(cafeId: String, castId: String?)
    case navigateToSchedule(castId: String?)
    case navigateToExternalLink(title: String, url: String)
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/cafemanagement/cafedashboard/CafeDashboardEvent.kt
package com.hhp227.concafe.presentation.main.cafemanagement.cafedashboard

sealed interface CafeDashboardEvent {
    data object NavigateBack : CafeDashboardEvent
    data object NavigateToBannerEdit : CafeDashboardEvent
    data class NavigateToCafeInfoEdit(val cafeId: String) : CafeDashboardEvent
    data class NavigateToNoticeEvent(val cafeId: String) : CafeDashboardEvent
    data class NavigateToMenuGoods(val cafeId: String) : CafeDashboardEvent
    data class NavigateToCastEdit(val cafeId: String, val castId: String? = null) : CafeDashboardEvent
    data class NavigateToSchedule(val castId: String? = null) : CafeDashboardEvent
    data class NavigateToExternalLink(val title: String, val url: String) : CafeDashboardEvent
}
